Add CONTRIBUTING, update README

phil-blain opened the pull request:
## PR checklist
- [x] Short (1 sentence) summary of your PR: 
    Add a Contribution guide (CONTRIBUTING.md) and update the README to make it more welcoming.
- [x] Developer(s): 
    Philippe Blain
- [x] Suggest PR reviewers from list in the column to the right.
I'll ping everyone on this one:
@eclare108213 @apcraig @duvivier @dabail10 @JFLemieux73 @proteanplanet @CICE-Consortium/devteam @mattdturner @rgrumbine @rallard77 @TillRasmussen 
- [x] Please copy the PR test results link or provide a summary of testing completed below.
    No tests.
- How much do the PR code changes differ from the unmodified code? 
    - [x] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [x] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [x] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [ ] Yes
    - [x] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [x] No 
- [x] Please provide any additional information or relevant details below:

This would close #384.

You can see the formatted documents over at my fork:
[CONTRIBUTING.md](https://github.com/phil-blain/CICE/blob/add-contributing/CONTRIBUTING.md)
[README.md](https://github.com/phil-blain/CICE/blob/add-contributing/README.md)

I would welcome any feedback: wording, formatting, section placements, etc.

I think the wording I chose makes it clear that support requests and feature requests should go to the bulletin board, but genuine bugs should be reported using GitHub issues. I think that is the consensus that was reached during our last call. If we feel like we should outright tell people to not open issues and always open a thread on the bulletin board instead, I'll update the PR.

I think our README looks a lot better and is more welcoming with these changes.

Please @-mention others I might have forgotten (especially new Consortium members as this is mostly targeted to them!)

P.S.: once consensus is reached on the content and wording I'll do a similar PR for Icepack.

apcraig commented:
I do think some of the content is quite helpful, but I also agree that we should not be duplicating links (or even content) in multiple places.  In fact, I think I've been the most proactive in that regard.  My preference would be that the only place we have links in on the resource page and all other places point ONLY to the resource page.  In the same way, documentation (like how to download the code or setup a case) would exist only in one place.  That way, we only have one thing to maintain when things change.

I think some of this content could be moved to the resource page which would almost certainly make it better.   Some of the info here could be dropped into the other documents as well.  I totally understand why it's appealing to do what's been done here, but I think it creates more work overall in some ways.

Again, my preference is that everything points to the resource page.  That resource page should be well organized and point to various documents and other things in a friendly and clear way.  And those documents on the resource page should overlap as little as possible (again to compartmentalize content and minimize redundancy).

eclare108213 commented:
To be clear @apcraig, are you suggesting to keep the basic outline of CONTRIBUTING but move the details elsewhere (especially the resources page)?  I think that could still be helpful.  @phil-blain where does CONTRIBUTING.md appear in GitHub?  Is it something that will be staring people in the face as soon as they open a repository page?

apcraig commented:
I guess what I'm suggesting is that both in the README and the CONTRIBUTING page that more or less the only link that exists (if possible) is to the CICE Resource Page.   We could then add/update the content on the CICE Resource Page (or other documents) based upon what @phil-blain has done here.  In other words, I propose both look something like

`This is the CICE README/CONTRIBUTING page.  For more information, see https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index which contains information about how to download and run the code, pointers to input data, how to get help, and many other things.`

But written with a little more fluff and pizazz.  If we wanted there to be quite a bit more information, that's OK.  I would just avoid having a bunch of links or redundant content if possible.  Again, that's just my proposal/preference for organizing our documentation.  The "half dozen" changes that were required to address the bulletin board change is a good example about why I think having everything centralized provides some benefits to us.  And if the resource page and our documentation is well organized, it should be more than adequate for users.

phil-blain commented:
Regarding the pull request template:
the new content is in HTML comments so it will not appear once the PR is created... I think our PRs looks more professionnal when the check boxes are correctly checked :P

but I can drop that commit if we feel it's not an issue.

apcraig commented:
I just want to ping this thread.  I think we should try to finish this up.  I think the main outstanding question is whether we want the README and CONTRIBUTING files to have so many details.  I have argued the README and CONTRIBUTING pages should point to the resource page for more details and we should not duplicate the documentation as that just creates additional maintenance work.  I think there are also other benefits if users just leverage the resource page as their go-to Consortium documentation/information starting point.

eclare108213 commented:
I agree that we should avoid duplication and link mainly (maybe only) to the resources page.  However that page does not provide a lot of guidance for someone approaching the Consortium for the first time.  Maybe the 'contributing' page can provide some textual direction without links that need to be maintained.  For example, we can say that the place to find information about dealing with the icepack submodule is in the git workflow document (which is linked from the resources page), and the place to find information about testing requirements for contributing new code, and information about the tests themselves, is in so-and-so section of the documentation (also linked from the resources page).  I don't think we should clutter up the resources page with a lot of extra, descriptive text.  This approach might 'feel' a little friendlier to users without being maintenance-heavy for us.  We'd still want to review it every once in a while.  I think the tutorial will provide us with some feedback on the kinds of guidance that new (and experienced) users might need most.

apcraig commented:
I think it's great to have some text to provide an overview.  I even don't mind having a few links (like to the resource page, bulletin board, and something else, for instance).  But I think we should consciously avoid too much redundancy.  I also think some of the updates in the README and CONTRIBUTING documents as they currently stand could/should be migrated to the resource page and/or other documents.  The information is nicely presented.  I think we should be constantly improving the resource page.  I feel it's one of the most critical pages in the Consortium project.

eclare108213 commented:
I'll be working on Consortium stuff for the tutorial this afternoon, traveling back to NM, and will try to take a stab at this text.

phil-blain commented:
I can also update this PR once a consensus has been reached... 

@eclare108213  to answer your question:
> @phil-blain where does CONTRIBUTING.md appear in GitHub? Is it something that will be staring people in the face as soon as they open a repository page?

When GitHub users go to open their first issue or pull request in a project, GitHub shows these banners:
For issues:
![issue_contributing](https://user-images.githubusercontent.com/44212482/73093141-eaedbc80-3eab-11ea-9c9c-42fb12e2fb53.png)
For PRs:
![pr_contributing1](https://user-images.githubusercontent.com/44212482/73093153-f0e39d80-3eab-11ea-8d2f-459f7ed3d574.png)

In both cases, the "contributing guidelines" link to CONTRIBUTING.md.

eclare108213 commented:
Can we control the text if those banners?
e


Sent with BlackBerry Work
(www.blackberry.com)


From: Philippe Blain <notifications@github.com<mailto:notifications@github.com>>
Date: Friday, Jan 24, 2020, 12:27 PM
To: CICE-Consortium/CICE <CICE@noreply.github.com<mailto:CICE@noreply.github.com>>
Cc: Hunke, Elizabeth Clare <eclare@lanl.gov<mailto:eclare@lanl.gov>>, Mention <mention@noreply.github.com<mailto:mention@noreply.github.com>>
Subject: [EXTERNAL] Re: [CICE-Consortium/CICE] Add CONTRIBUTING, update README and PULL_REQUEST_TEMPLATE (#392)


I can also update this PR once a consensus has been reached...

@eclare108213<https://github.com/eclare108213> to answer your question:

@phil-blain<https://github.com/phil-blain> where does CONTRIBUTING.md appear in GitHub? Is it something that will be staring people in the face as soon as they open a repository page?

When GitHub users go to open their first issue or pull request in a project, GitHub shows these banners:
For issues:
[issue_contributing]<https://user-images.githubusercontent.com/44212482/73093141-eaedbc80-3eab-11ea-9c9c-42fb12e2fb53.png>
For PRs:
[pr_contributing1]<https://user-images.githubusercontent.com/44212482/73093153-f0e39d80-3eab-11ea-8d2f-459f7ed3d574.png>

In both cases, the "contributing guidelines" link to CONTRIBUTING.md.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/CICE-Consortium/CICE/pull/392?email_source=notifications&email_token=AB6LTH3W2XSSPKAAQNAFZKTQ7MXINA5CNFSM4KD7BVZKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEJ3VIEQ#issuecomment-578245650>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AB6LTH6BND7B5QGLM55GNI3Q7MXINANCNFSM4KD7BVZA>.

phil-blain commented:
No.

phil-blain commented:
> I would like to suggest that we use the term "forum" instead of "bulletin board", everywhere it appears. The latter sounds like it's just for announcements, while "forum" invites discussion.

fixed

> Re README
> I've tried not to emphasize any Consortium institution in the first-look info, including LANL, which is why that first statement isn't in the README now. 

fixed

phil-blain commented:
@apcraig 
> I would remove the new content in the PULL_REQUEST_TEMPLATE.

done

phil-blain commented:
@eclare108213 
> I would prefer that issues be reserved for Consortium member use -- have users go first to the forum, and we (or they, with our permission) can create an issue after we've had a chance to review the problem. 

done.
> I would probably reword the porting bullet to emphasize that they should probably involve their IT people first!

yes, good idea indeed! done.

eclare108213 commented:
Here are my _suggestions_.  I started from the versions 4+ hours ago and didn't do a careful comparison, so apologies for not capturing your latest commits, @phil-blain.

Thank you for changing to "forum" from "bulletin board" -- we need to do this everywhere it appears, throughout the repos. 

**README**

Add at the top of the original (This is mostly there now in your version, but I edited it slightly):

CICE is a computationally efficient model for simulating the growth, melting, and movement of polar sea ice. Designed as one component of coupled atmosphere-ocean-land-ice global climate models, today’s CICE model is the outcome of more than two decades of community collaboration in building a sea ice model suitable for multiple uses that include process studies, operational forecasting, and climate simulation.

Replace 

>If you expect to make any changes to the code, we recommend that you first fork both the CICE and Icepack repositories. Basic instructions for working with CICE and Icepack are found in the Git Workflow Guidance, linked from the Resource Index (below). In order to incorporate your developments into the Consortium code it is imperative you follow the guidance for Pull Requests and requisite testing.

With

>If you need help getting started using the model after reviewing the model documentation, the first point of contact with the CICE Consortium is the Consortium Community Forum (linked below). This forum is monitored by Consortium members and open to the whole community. Please do not use our issue tracker for general support questions.
>
>If you expect to make any changes to the code, we recommend that you first fork both the CICE and Icepack repositories. In order to incorporate your developments into the Consortium code it is imperative you follow the guidance for Pull Requests and testing. Head over to our Contribution guide to learn more about how you can help improve CICE.

And link the contribution guide appropriately.  This link should also be added to the Resources Index.

**CONTRIBUTING**

Rather than cloning @phil-blain's wiki branch for this PR, I will just paste my entire, edited draft into the following comment box.  You can 'edit' the comment to grab the source text.  We can continue to edit it here if you/we think this version is close to final.  I'm open to criticism!

This draft is a compromise. I reduced the number of links to maintain, keeping links to the most critical information on the wiki and some into the repository that aren’t likely to change.  I've been working offline (at 30,000 feet) and so we need to make sure that the things now referred to are quickly and obviously available in the remaining links and/or Resources index.  E.g. is the software development guide linked at the top of the developer_guide?  If not, is it linked on the Resources index?  Do the links to the developer and user guides bring up a table of contents that contain obvious internal links to the desired information?  If not, we could be a little more specific about which section to look in, with or without linking to it.

eclare108213 commented:
# Contributing to CICE

First, thanks a lot for considering a contribution to CICE!

## Getting help
The central index for CICE repository information is the [Resources][resources] page, which contains links to code, documentation, testing information, guidelines, and more.  

If you still need help getting started using the model after reviewing the model documentation, the first point of contact with the CICE Consortium is the [Consortium Community Forum][forum]. This forum is monitored by Consortium members and is also open to the whole community for input.
**Please do not use our issue tracker for general support questions.**

[resources]: https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index
[forum]: https://xenforo.cgd.ucar.edu/cesm/forums/cice-consortium.146/

## Reporting issues
We use GitHub issues mainly for confirmed bugs in the model and discussion of Consortium-led development tasks. If you believe you have found a bug in CICE, please open a thread on the [Forum][forum]. We will create a GitHub issue if needed.  In your thread, be sure to:
- Document (1) the steps you took, (2) the expected behavior of the model and (3) its actual behavior
- Identify the [supported machine](/configuration/scripts/machines) on which you are running CICE
- If you are running on an unsupported machine and have a porting related problem:
   - Talk with your local IT help desk.  They will likely be more help than we can be.
   - If you are still stuck, then let us know your operating system and version, and provide the `env.<machine>_<compiler>` and `Macros.<machine>_<compiler>` files you are using.

For feature/enhancement proposals, we suggest first opening a [new thread][new-thread] on the [Forum][discussion-board-wiki].

[new-thread]: https://xenforo.cgd.ucar.edu/cesm/forums/cice-consortium.146/post-thread
[discussion-board-wiki]: https://github.com/CICE-Consortium/About-Us/wiki/Contacting-the-Consortium#discussion-board

## Contributing to development
New CICE features should first be discussed with the Consortium team members in the [Forum][forum]. 

A quick-start overview of how to get the code, compile the model and run a case is available in the documentation, linked from the [Resources][resources] page.

Please review and follow our Software Development Practices when contributing code. 
To get started coding, take a look at the [Developer Guide][developer-guide] section of the model documentation.

[developer-guide]: https://cice-consortium-cice.readthedocs.io/en/master/developer_guide/index.html

### Version control workflow
CICE uses a forking and Pull Request (PR) workflow, so start by [forking][cice-fork] the repo to your GitHub account. We use feature branches for development, so please keep your pull requests self-contained, i.e. your pull request branch should be separate from `master` and should not contain unrelated changes.  

A detailed overview of our [Git workflow][git-workflow], including a Git primer for less experienced users, is found on the [About-Us wiki][about-us].

The column portion of the model, [Icepack][icepack], is included in CICE as a Git submodule. Submodules are a more advanced feature of Git that even experienced Git users might not have encountered, and we recommend getting comfortable with the concept.
Some reading materials, in increasing level of complexity:
- The [Submodules section][git-tower] of the Git Tower tutorial (introductory)
- The [Submodules chapter][git-book-submodule] of the [Pro Git][pro-git] book (in depth)
- The official Git documentation: [gitsubmodules][gitsubmodules], [git-submodule][git-submodule], [gitmodules][gitmodules] (reference)

We strongly recommend you review the [Git workflow][git-workflow] page, which describes how to work with the Icepack submodule.  

[cice-fork]: https://github.com/CICE-Consortium/CICE/fork
[git-workflow]: https://github.com/CICE-Consortium/About-Us/wiki/Git-Workflow-Guidance
[about-us]: https://github.com/CICE-Consortium/About-Us/wiki
[icepack]: https://github.com/CICE-Consortium/Icepack
[git-book-submodule]: https://git-scm.com/book/en/v2/Git-Tools-Submodules
[pro-git]: https://git-scm.com/book/en/v2
[git-tower]: https://www.git-tower.com/learn/git/ebook/en/command-line/advanced-topics/submodules
[gitsubmodules]: https://git-scm.com/docs/gitsubmodules
[git-submodule]: https://git-scm.com/docs/git-submodule
[gitmodules]: https://git-scm.com/docs/gitmodules

### Testing
Any contribution to CICE must be properly tested to ensure that code modifications do not unexpectedly change the model output.  This is a prerequisite for a PR to be reviewed.  The required level of testing is described briefly in our Software Development Practices, and more in depth in the testing section of the [User Guide][user-guide].  A detailed walk-through is available in the End-To-End Testing Procedure section of the User Guide.  

PRs introducing new features should have corresponding tests added to the [base_suite](configuration/scripts/tests/base_suite.ts).

[user-guide]: https://cice-consortium-cice.readthedocs.io/en/master/user_guide/index.html

### Documentation
We often hear from users that the CICE code is great but the documentation is _awesome_.  Please help us keep it that way by letting us know about weaknesses or errors (through the [Forum][forum]) and by providing complete documentation for any new code features. 

The source of the CICE model documentation is in the [`doc/source`](doc/source) folder of the repository, and is written in [reStructuredText][rst] using [Sphinx][sphinx].  To make documentation updates easier, opening a pull request triggers a build of the associated documentation at `https://external-builds.readthedocs.io/html/cice-consortium-cice/<pr-number>/index.html`, so we can read the new, formatted docs when reviewing the PR. Updates to the `master` branch when PRs are merged automatically update [Read the Docs][rtd].  

The [About-Us wiki][wiki-doc] has information about reStructuredText syntax, building the documentation locally, and Frequently Asked Questions about the documentation.

[rst]: https://docutils.sourceforge.io/rst.html
[sphinx]: http://www.sphinx-doc.org/en/master/
[rtd]: https://cice-consortium-cice.readthedocs.io/
[wiki-doc]: https://github.com/CICE-Consortium/About-Us/wiki/Documentation-Workflow-Guidance

### Code review and merging process
Your pull request will be reviewed by members of the Consortium. You might be asked to polish your PR, add tests or modify documentation before it is merged into the `master` branch. Please respond to this feedback and feel free to ask questions. Once the reviewers are satisfied with the pull request, it will be [squashed-merged][squash] into the `master` branch. Any forward development at this point should happen in a _new_ branch.

A helpful hint: Use [GitHub Flavored Markdown][gfm] appropriately, especially if you include [preformatted text][code].

[squash]: https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges#squash-and-merge-your-pull-request-commits
[gfm]: https://help.github.com/en/github/writing-on-github
[code]: https://help.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks

apcraig commented:
Personally, I think this still has way too many links and way too many details.  I would prefer to see a lot of this rolled into other documentation or onto the resources page.  But that's just my opinion.  Very open if others feel differently.

I'm happy to help migrate some of this content into other documents, but don't want to unilaterally do anything without general agreement about strategy first.

eclare108213 commented:
A lot of the links are to resources that are not linked anywhere else in our repos or wikis.  The other links are these:

On wiki or external:
Resources index
User guide
Developer guide
Git workflow
About-Us
About-Us (docs)
Forum
Readthedocs

In repo:
Icepack
Base_suite
Doc/source

This document provides context for all of these things for someone approaching the Consortium for the first time, and makes them aware of what is available.  It definitely points people to the primary resources, which are the resources index, the documentation, and the forum. We can remove some of the others, but I thought a little finer grain on the documentation resources could be helpful (message:  it's important!).  We know that no one is going to sit down and read through all of the documentation to get a broad overview, even of the major sections -- people tend to go there looking for something specific, and this lets them know that there is a _lot_ there.

As I said, I'm also open to doing this differently.  I do like the "friendly" introduction that this provides.

apcraig commented:
I would argue the following.  

- This will largely have to be duplicated in Icepack as well, so now we have even more redundant links.  And the README and CONTRIBUTING pages all have redundant links between them.
- With this first introduction, we're pointing folks to About-Us, external documentation (not produced by us), some of our own internal documentation, the resource page, the Forum, and so forth.  That could create some confusion about how our documentation is organized.  I think we want people to realize our resource page is the primary place to go.
- Or why can't this page become our resource page (more or less).  
- Or maybe convert this page to a "Welcome to the Consortium Page" that the Icepack and CICE README and CONTRIBUTING pages point to (after a paragraph of introductory text).  We then need to make sure we have a clear section that says, all the documentation and resource information can be found on the resource pages.
- The resource page can continue to improve.

Again, I'd like for any place a user could land to point to the resource page (or maybe sometimes to a higher level welcome page that the above content could become but that also points to the resource page).  And then the resource page points out to all the resources.  Ideally, we should be able to create a very simply dependency graph for our documentation and links.  As currently implemented, we don't quite meet my goal for documentation organization, and I haven't been overly aggressive (I hope) about pushing for it.  But the current expansion of the README and CONTRIBUTING pages is working very much away from that goal, and is something I'd prefer we not do.  Again, the problem is not the content (which is great), it's the organization.

Again, happy to defer, but if this content is valuable, it should be in a more centralized location and not just appearing the first time someone clicks on the consortium and never again.

eclare108213 commented:
These are good points and suggestions.  It makes sense to me to make the CONTRIBUTING files for both CICE and Icepack more or less identical and very brief.  They could point to the essential information linked at the bottom of the readme (i.e. link to the readme) and also link to a single file that contains the info about, perhaps pared down a bit, located on the About-Us wiki (right next to the Resources page).  I like that the Resources page is an index without a lot of extra verbiage -- it's easy to find stuff on there -- and I like that this page is more conversational.  I'd be fine not linking with Sphinx or the main readthedocs page, and only naming sections of the documentation without linking to them.  I do think it's helpful to still link (again) to the main resources, like the git workflow, but can be convinced otherwise.  And I think this page, wherever it is, should be linked from the Resource index.  I'm hoping others will weigh in...  @duvivier  ?  I'll take a stab at a generic welcome message and we can iterate.

eclare108213 commented:
What about something like this for CONTRIBUTING.md in both CICE and Icepack?  On that page, I'd still keep the basic information available above, but instead of links that reach into various parts of the documentation, perhaps we could make a diagram that shows where to find it and/or how it's laid out.  I'm concerned that new users will spin their wheels trying to find something they don't quite know how to look for, or that they won't know to look for it to begin with.   

> Welcome to the CICE Consortium!  There are many ways to contribute, and we appreciate your efforts.  To get started, take a look at the [Resource Index][resources], our one-stop shop for Consortium information.  For an introductory tour, head over to our [Contribution Guide]LINK. 

[resources]: https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index

apcraig commented:
I think the proposed paragaph is a good one.  I don't mind if there is a bit more content such as saying we leverage git, we have user guides and developer guides, we have lots more resources, etc.  I would also suggest you swap the order of the links.  I would suggest the general order be

Welcome ....   The Consortium is blah blah blah and more blah blah blah.  A general introduction and great starting point to learn more about the project is [Contribution Guide].  While our [Resources Page] is the goto link for all documentation, internal and external links, and other information.

eclare108213 commented:
Keeping in mind that this is what will be linked from the one-time banners that @phil-blain shows above, I wouldn’t include too much blah blah.  Switching the order makes sense. How about

For an introductory tour, head over to our [Contribution Guide]LINK, and take a look at the [Resources Index], our go-to link for all....

apcraig commented:
That looks good to me.  Thanks.

eclare108213 commented:
I added a new [Contributing](https://github.com/CICE-Consortium/About-Us/wiki/Contributing) page to the About-Us wiki but did not create any links to it except this one.  I further reduced the internal links and took out some but not all of the external links.  Please take a look and make suggestions in this PR conversation. 

Here are my suggestions for other pages:

**CONTRIBUTING.md**

> Welcome to the CICE Consortium! There are many ways to contribute, and we appreciate your efforts.  For an introductory tour, head over to our [Contribution guide] and take a look at the [Resources page], our go-to index with links to code, documentation, testing information, guidelines, and more.

**Resource index**

- Change ‘bulletin board’ to ‘forum’  (do this on other pages where it appears)
- Under Model Testing, add a link ‘See also the [Information for Developers] below.’
- Currently, ‘Contributing to model development’ links to the FAQ.  Instead, this should link to the new 'Contributing' wiki page and the FAQ should also point to the new page.  

**FAQ**
- Link to 'Contributing' wiki page
- The testing criteria on the FAQ should be removed and instead linked to the Required Testing section of the Software Development Practices guide. 
- (Note to self: update the institutional member list on the FAQ, or remove it)

Top of **README** (include links to CICE-svn-trunk and at the bottom as before, but use ‘forum’ instead of ‘bulletin board’):

>CICE is a computationally efficient model for simulating the growth, melting, and movement of polar sea ice. Designed as one component of coupled atmosphere-ocean-land-ice global climate models, today’s CICE model is the outcome of more than two decades of community collaboration in building a sea ice model suitable for multiple uses that include process studies, operational forecasting, and climate simulation.
>
>This repository contains the files and code needed to run the CICE sea ice numerical model starting with version 6. CICE is maintained by the CICE Consortium. Versions prior to v6 are found in the [CICE-svn-trunk](https://github.com/CICE-Consortium/CICE-svn-trunk) repository.
>
>CICE consists of a top level driver and dynamical core plus the Icepack column physics code, which is included in CICE as a git submodule. Because Icepack is a submodule of CICE, Icepack and CICE development are handled independently with respect to the github repositories even though development and testing may be done together.
>
>If you need help getting started using the model after reviewing the model documentation, the first point of contact with the CICE Consortium is the Consortium Community Forum (linked below). This forum is monitored by Consortium members and open to the whole community. Please do not use our issue tracker for general support questions.
>
>If you expect to make any changes to the code, we recommend that you first fork both the CICE and Icepack repositories. In order to incorporate your developments into the Consortium code it is imperative you follow the guidance for Pull Requests and testing. Head over to our Contribution Guide (linked from the Resource Index below) to learn more about how you can help improve CICE.

apcraig commented:
I like the new contributing page.  I would suggest a few changes.  I would try to avoid the "About-Us" references/links.  I view the About-Us as sort of a potential landing page and a place where we keep a lot of our documentation.  But I don't think of it as a place that should be linked to or really mentioned.  I don't think there is anything on the About-Us page that is not on the Resource Page.  And I think we should try to avoid mentioning anything related to the organization of our documentation, so the fact that a bunch of the documentation is on the About Us wiki vs the Icepack wiki or CICE wiki or even external should not be exposed.  So, I would change

> A detailed overview of our [Git workflow], including a Git primer for less experienced users, is found on the [About-Us] wiki.
> 
> to 
> 
> We provide a detailed overview of our [Git workflow] including a Git primer for less experienced users.
> 

I would change 

>  We even have documentation about the documentation: the [About-Us wiki] has information about syntax, building the documentation locally, and Frequently Asked Questions.
> 
> to 
> 
>  We even have documentation about the documentation: the [Documentation Workflow Document] has information about syntax, building the documentation locally, and Frequently Asked Questions.
> 

I would not mention the squashed merge at the end.  And I don't fully understand what the final sentence/link is contributing. (A helpful hint: Use [GitHub Flavored Markdown] appropriately.)

Finally, I would move this

> The column portion of the model, Icepack, is included in CICE as a Git submodule. Submodules are a more advanced feature of Git that even experienced Git users might not have encountered, and we recommend getting comfortable with the concept. Some reading materials, in increasing level of complexity:
> 
>     The Submodules section of the Git Tower tutorial (introductory)
>     The Submodules chapter of the Pro Git book (in depth)
>     The official Git documentation: gitsubmodules, git-submodule, gitmodules (reference)
> 

to the git workflow document under submodules.  I think noting submodules are important is good, but I think pointing to the git workflow document is adequate at this level.  If we include these links in the submodule section of the workflow document, I think that would be best.

eclare108213 commented:
I like these suggestions @apcraig and have made the changes to the 'contributing' wiki page, including moving the submodule links to the git workflow doc.  I have not made the other changes that I suggested above, waiting for others' opinions.

phil-blain commented:
@eclare108213 : I think having the Contributing page in the About-Us wiki is a good compromise. It's true that it causes less administrative burden to update and less duplication since CICE and Icepack can both link to it instead of duplicating content. 
To give more visibility to this page, I would link to it from a few places: 
- the home page of the About-Us wiki https://github.com/CICE-Consortium/About-Us/wiki
- the side bar of the About-Us wiki  (super important to make it easy to find)
- CONTRIBUTING.md in CICE and Icepack
- probably also README.md in CICE and Icepack

I'm also ok with moving the submodule resources to the Git Workflow page.

---

Regarding the comments by @apcraig :
>  I would try to avoid the "About-Us" references/links. 

The reason why I explicitly mentioned "the About-Us wiki" is that when I started with the Consortium, I was always confused by how the documentation is organized. For example, I would try to find the "Git workflow Guidance" page often, first by looking in the CICE wiki, not finding it, then clicking Resource Index, looking at the sidebar, still not finding it, then scrolling down and finally finding the link. 
Or I would go to the About-Us repo, and then not find it either.
So that is why I think it's good to name things, then people might remember more easily where they are.

> I would not mention the squashed merge at the end. 

For some contributors that are used to make [atomic](https://fnune.com/2018/02/19/git-best-practices-atomic-commits/), self-contained commits, and write really detailed commit messages, it can be quite disconcerting for their pull requests to be integrated using this procedure, because all the work they did in splitting their work into separate commits is discarded. So I think we should at least warn these users that that is the procedure that is going to be used.

> And I don't fully understand what the final sentence/link is contributing. (A helpful hint: Use [GitHub Flavored Markdown] appropriately.)

I agree; I had added that sentence when I talked about opening issues. In the latest version of this branch, I instead link to the appropriate syntax page on xenodo when I talk about opening new forum threads.

---

@eclare108213 regarding your suggestions to the different files/pages:

**CONTRIBUTING.md**: maybe I'd copy the "Getting help" section in there, so that it may deter more people from creating issues (since they have less links to click to in order to learn that we prefer they open a thread in the Forum)

**FAQ and Resource Index**: I think these are good ideas.

**README**: I would link directly in the text to the Forum, as well as to the Contributing page on the About-Us wiki.

---
Also of note: I just noticed that the About-Us *repo* has a code of conduct https://github.com/CICE-Consortium/About-Us/blob/master/CodeOfConduct.md
That is super, and we should really link to that in the new Contributing page of the wiki.

duvivier commented:
Sorry I'm late to the thread here. @phil-blain, thanks for heading this up. I didn't know about this feature and it will be good to follow the expected github format of repos.

I've read through the comments so far and here are my opinions:

- I like the more detailed descriptions throughout, but I am also concerned like @apcraig with how many links there are in here that will be challenging to keep track of. This was the whole purpose of creating the resource links page - to minimize how messy links can get.

- If we keep a more verbose contributing.md format, I think that we should then keep that version in the About-Us repo. Then the contributing.md files in CICE and Icepack could both point to the contributing.md in About-Us. 

@phil-blain I also don't like how the pull request template check boxes don't match up with the numbering. But I don't know how to fix that given that we don't have a linear PR process since we can have, for example, both b4b and non requests. But that information seems essential to me in the PR template so we can adequately review it. 

I should be online all tomorrow to get this settled before the tutorial if that's our timeframe before people all start looking at this.

eclare108213 commented:
That's great!  Thank you.

e

________________________________
From: Alice DuVivier <notifications@github.com>
Sent: Friday, January 31, 2020 12:04:39 AM
To: CICE-Consortium/CICE
Cc: Hunke, Elizabeth Clare; Mention
Subject: [EXTERNAL] Re: [CICE-Consortium/CICE] Add CONTRIBUTING, update README (#392)


Sorry I'm late to the thread here. @phil-blain<https://github.com/phil-blain>, thanks for heading this up. I didn't know about this feature and it will be good to follow the expected github format of repos.

I've read through the comments so far and here are my opinions:

  *   I like the more detailed descriptions throughout, but I am also concerned like @apcraig<https://github.com/apcraig> with how many links there are in here that will be challenging to keep track of. This was the whole purpose of creating the resource links page - to minimize how messy links can get.

  *   If we keep a more verbose contributing.md format, I think that we should then keep that version in the About-Us repo. Then the contributing.md files in CICE and Icepack could both point to the contributing.md in About-Us.

@phil-blain<https://github.com/phil-blain> I also don't like how the pull request template check boxes don't match up with the numbering. But I don't know how to fix that given that we don't have a linear PR process since we can have, for example, both b4b and non requests. But that information seems essential to me in the PR template so we can adequately review it.

I should be online all tomorrow to get this settled before the tutorial if that's our timeframe before people all start looking at this.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/CICE-Consortium/CICE/pull/392?email_source=notifications&email_token=AB6LTH3DFFSQ4I3SM4WE3QDRAPEQPA5CNFSM4KD7BVZKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEKNWU4Q#issuecomment-580610674>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AB6LTH7TZT7HQFTCSSZKLQ3RAPEQPANCNFSM4KD7BVZA>.

eclare108213 commented:
Whoops, I responded to the wrong email... sorry about that.

e

________________________________
From: Hunke, Elizabeth Clare
Sent: Friday, January 31, 2020 5:37:52 AM
To: CICE-Consortium/CICE; CICE-Consortium/CICE
Cc: Mention
Subject: Re: [EXTERNAL] Re: [CICE-Consortium/CICE] Add CONTRIBUTING, update README (#392)


That's great!  Thank you.

e

________________________________
From: Alice DuVivier <notifications@github.com>
Sent: Friday, January 31, 2020 12:04:39 AM
To: CICE-Consortium/CICE
Cc: Hunke, Elizabeth Clare; Mention
Subject: [EXTERNAL] Re: [CICE-Consortium/CICE] Add CONTRIBUTING, update README (#392)


Sorry I'm late to the thread here. @phil-blain<https://github.com/phil-blain>, thanks for heading this up. I didn't know about this feature and it will be good to follow the expected github format of repos.

I've read through the comments so far and here are my opinions:

  *   I like the more detailed descriptions throughout, but I am also concerned like @apcraig<https://github.com/apcraig> with how many links there are in here that will be challenging to keep track of. This was the whole purpose of creating the resource links page - to minimize how messy links can get.

  *   If we keep a more verbose contributing.md format, I think that we should then keep that version in the About-Us repo. Then the contributing.md files in CICE and Icepack could both point to the contributing.md in About-Us.

@phil-blain<https://github.com/phil-blain> I also don't like how the pull request template check boxes don't match up with the numbering. But I don't know how to fix that given that we don't have a linear PR process since we can have, for example, both b4b and non requests. But that information seems essential to me in the PR template so we can adequately review it.

I should be online all tomorrow to get this settled before the tutorial if that's our timeframe before people all start looking at this.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/CICE-Consortium/CICE/pull/392?email_source=notifications&email_token=AB6LTH3DFFSQ4I3SM4WE3QDRAPEQPA5CNFSM4KD7BVZKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEKNWU4Q#issuecomment-580610674>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AB6LTH7TZT7HQFTCSSZKLQ3RAPEQPANCNFSM4KD7BVZA>.

eclare108213 commented:
We are converging on this issue and I think we need to finish it up today.  Remaining questions are in @phil-blain s comments above.  I'm generally okay with all of those suggestions, but I think the jury is still out on where and how often to link the various pages.  I like the idea of just linking to the resources page in theory, but in practice I can see how that is non-intuitive to people approaching the Consortium for the first few times.  I think we should go with the single contributing page in About-Us (without too many links within it), briefly link to it from CONTRIBUTING.md (and we can add the 'how to get help' section there, but I might not), and then link to the contributing wiki from the READMEs.  I don't think a contributing link needs to be explicit in the side bar, but then again I also find it a little frustrating to look there and not find stuff.  Maybe we can annotate the sidebar link to 'resources' to make it more obvious as the place to go?

duvivier commented:
I'm answering Phil's questions directly from his last message. I think this can be done today pretty easily. @phil-blain , if I can help with any of these last items let me know.

> @eclare108213 : I think having the Contributing page in the About-Us wiki is a good compromise. It's true that it causes less administrative burden to update and less duplication since CICE and Icepack can both link to it instead of duplicating content.
> To give more visibility to this page, I would link to it from a few places:
> 
> * the home page of the About-Us wiki https://github.com/CICE-Consortium/About-Us/wiki
> * the side bar of the About-Us wiki  (super important to make it easy to find)
> * CONTRIBUTING.md in CICE and Icepack
> * probably also README.md in CICE and Icepack
> 

I'm fully on board with this solution. Linking in all the places you list, @phil-blain , makes sense to me.

> I'm also ok with moving the submodule resources to the Git Workflow page.
> 
This sounds good to me.

> Regarding the comments by @apcraig :
> 
> > I would try to avoid the "About-Us" references/links.
> 
> The reason why I explicitly mentioned "the About-Us wiki" is that when I started with the Consortium, I was always confused by how the documentation is organized. For example, I would try to find the "Git workflow Guidance" page often, first by looking in the CICE wiki, not finding it, then clicking Resource Index, looking at the sidebar, still not finding it, then scrolling down and finally finding the link.
> Or I would go to the About-Us repo, and then not find it either.
> So that is why I think it's good to name things, then people might remember more easily where they are.
> 
I think this is good feedback that the process wasn't intuitive at first. If non team members aren't sure where to look and would usually look for a "contributing.md" file first for guidance, then we should create that with the info in it and links to other places (like the About-Us wiki). I think we can try this solution and we can always refine later, if necessary. 

> > I would not mention the squashed merge at the end.
> 
> For some contributors that are used to make [atomic](https://fnune.com/2018/02/19/git-best-practices-atomic-commits/), self-contained commits, and write really detailed commit messages, it can be quite disconcerting for their pull requests to be integrated using this procedure, because all the work they did in splitting their work into separate commits is discarded. So I think we should at least warn these users that that is the procedure that is going to be used.
> 
I don't have as much familiarity with this, so I don't have an opinion to share.

> > And I don't fully understand what the final sentence/link is contributing. (A helpful hint: Use [GitHub Flavored Markdown] appropriately.)
> 
> I agree; I had added that sentence when I talked about opening issues. In the latest version of this branch, I instead link to the appropriate syntax page on xenodo when I talk about opening new forum threads.
> 
> @eclare108213 regarding your suggestions to the different files/pages:
> 
> **CONTRIBUTING.md**: maybe I'd copy the "Getting help" section in there, so that it may deter more people from creating issues (since they have less links to click to in order to learn that we prefer they open a thread in the Forum)
> 
Sounds good to me.

> **FAQ and Resource Index**: I think these are good ideas.
> 
Yep, agree.

> **README**: I would link directly in the text to the Forum, as well as to the Contributing page on the About-Us wiki.
> 
Yep, agree.

> Also of note: I just noticed that the About-Us _repo_ has a code of conduct https://github.com/CICE-Consortium/About-Us/blob/master/CodeOfConduct.md
> That is super, and we should really link to that in the new Contributing page of the wiki.
FYI, I'm opening the workshop and tutorial with this code of conduct info (as well as other logistics). This is an NCAR policy for meetings and I think just a good precedent to set.

Again, @phil-blain just let me know today what I can do to help. Reviewing, or helping with some of the PR's in other repos, etc. If you have it covered, then that's fine too. Thanks for heading this up! :)

apcraig commented:
I'm happy to help push this forward today as well.  I think we're converging and are reaching a reasonable compromise.  As @duvivier says, we can always revise again in the future.  It will be helpful to have this merged, then we can actually see what it looks like and continue to improve.  Do we have a clear list of things that still needs to be done?  I think it's at least

- update Contributing.md - simple paragraph pointing to contributing.md page as suggested by @eclare108213 above
- update README.md - reduce again as suggested by @eclare108213 above
- links to Contributing.md from several places
- move submodule resources to github workflow doc

On the squashed merge, I'm OK if others think it should be mentioned, but I don't think it should be one of the first things a user sees.  Can we drop it into the github workflow doc as well.  

I will update the github workflow doc now.  I think for this PR, we mainly need to finalize the Contributing and README .md files.  Those should be fairly straight-forward, largely pointing just to some primary links (resources, contributing, forum, etc).  I think @eclare108213 has provided a reasonable version for both.  By making them simple, they should be fairly static.  We can then continue to improve all the surrounding documentation without needing to PR.

phil-blain commented:
OK I'll update the README and CONTRIBUTING files soon.

apcraig commented:
I have updated the git workflow document and confirm the submodule links are there.  I also copied the squash-merge info into the PR section,

https://github.com/CICE-Consortium/About-Us/wiki/Git-Workflow-Guidance#submodules
https://github.com/CICE-Consortium/About-Us/wiki/Git-Workflow-Guidance#pull-request

I also added a link to the Contributing Page on the About-Us README and on the About-Us wiki sidebar

https://github.com/CICE-Consortium/About-Us
https://github.com/CICE-Consortium/About-Us/wiki

phil-blain commented:
@eclare108213 @apcraig @duvivier :

I've pushed a commit reducing the length of CONTRIBUTING.md and instead linking to the About-Us wiki Contributing page, as discussed above.

This PR should be ready to merge, but we should make sure to do all the things we mentioned above that pertain to other repos (or check they have been done): 
- [ ] change resource index as @eclare108213 wrote here https://github.com/CICE-Consortium/CICE/pull/392#issuecomment-578412376
- [ ] change FAQ @eclare108213 wrote here https://github.com/CICE-Consortium/CICE/pull/392#issuecomment-578412376
- [ ] do an icepack PR with README and CONRBUTING similar to the CICE ones - I can do that as soon as this one is approved
- [ ] make sure all feedback from this PR has been incorporated in the Contributing page on the wiki
- [ ] link to the code of conduct from the Contributing on the wiki to give it more visibility ?

duvivier commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,17 @@
+# Contributing to CICE
+
+Welcome to the CICE Consortium ! First, thanks a lot for considering to contribute to CICE.
```
Nit picky thing, sorry. Change to:

Welcome to the Consortium! Thank you for your interest in contributing to CICE.

eclare108213 replied:
done

duvivier commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,17 @@
+
+Welcome to the CICE Consortium ! First, thanks a lot for considering to contribute to CICE.
+
+There are many ways to contribute, and we appreciate your efforts. For an introductory tour, head over to our [Contribution guide][contributing] and take a look at the [Resources page], our go-to index with links to code, documentation, testing information, guidelines, and more.
```
It doesn't appear the resources page is linked. Another thing that @eclare108213 has also suggested in the past is including a full URL instead of just a link. We could consider that too.

eclare108213 replied:
linked to resources page, did not include full URLs

duvivier commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,17 @@
+[resources]: https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index
+
+## Getting help
+If you still need help getting started using the model afer reviewing the [model documentation][doc-running], the first point of contact with the CICE Consortium is the [Consortium Community Forum][forum]. 
```
My suggestion would be to link the documentation to the Resources Page "Model Documentation" section rather than directly into RTD. Sometimes RTD does weird, unexpected things so I think being one step removed would be preferable.

https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#model-documentation

apcraig replied:
There is a type here "afer".

duvivier commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,17 @@
+
+## Getting help
+If you still need help getting started using the model afer reviewing the [model documentation][doc-running], the first point of contact with the CICE Consortium is the [Consortium Community Forum][forum]. 
+This forum is monitored by Consortium members and also opened to the whole community.
```
I'd just say "... also open to the whole..." instead of "opened"

duvivier commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,17 @@
+## Getting help
+If you still need help getting started using the model afer reviewing the [model documentation][doc-running], the first point of contact with the CICE Consortium is the [Consortium Community Forum][forum]. 
+This forum is monitored by Consortium members and also opened to the whole community.
+**Please do not use our issue tracker for general support questions.**
```
Maybe have this on its own line?

apcraig replied:
I might even exclude this point of information.  Lets wait to see if we have problems with it before we say don't do it.  We point people to the forum early on, they should not end up on the issues page.

eclare108213 replied:
I can go either way on this point.  It's not a bad idea to make it very explicit because we've had quite a number of people submitting issues unnecessarily in the past.  Of course, they might not read this page so the problem will likely not ever go away completely.

eclare108213 replied:
I decided to leave the statement in place.

duvivier commented on README.md:
```diff
@@ -1,13 +1,32 @@
+Versions prior to v6 are found in the [CICE-svn-trunk repository](https://github.com/CICE-Consortium/CICE-svn-trunk).
+
+CICE consists of a top level driver and dynamical core plus the [Icepack column physics code][icepack], which is included in CICE as a Git submodule.  
+Because Icepack is a submodule of CICE, Icepack and CICE development are handled independently with respect to the GitHub repositories even though development and testing may be done together.  
```
The spacing her with the rendered document is a bit odd. Either add a space between the part that starts with "CICE consists..." and the part that starts with "Because Icepack..." otherwise, make it a continuous paragraph.

eclare108213 replied:
fixed

duvivier commented on README.md:
```diff
@@ -1,13 +1,32 @@
+[icepack]: https://github.com/CICE-Consortium/Icepack
+
+## Getting help
+If you need help getting started using the model afer reviewing the [model documentation][doc-running], the first point of contact with the CICE Consortium is the [Consortium Community Forum][forum]. 
```
Same comment about where documentation should link to as the contributing.md

eclare108213 replied:
done

duvivier commented on README.md:
```diff
@@ -1,13 +1,32 @@
 
-## Overview
-This repository contains the files and code needed to run the CICE sea ice numerical model starting with version 6. CICE is maintained by the CICE Consortium. Versions prior to v6 are found in the [CICE-svn-trunk repository](https://github.com/CICE-Consortium/CICE-svn-trunk).
+## The CICE Consortium sea-ice model
```
@eclare108213 may want to weigh into how we refer to this model. My inclination would be to label this section "The CICE sea-ice model"

eclare108213 replied:
I agree it's more natural to just say "CICE sea-ice model"

eclare108213 replied:
I've left this as 'The CICE Consortium sea-ice model' for now, but we can change it if others think that sounds weird.

duvivier commented on README.md:
```diff
@@ -18,10 +37,13 @@ imperative you follow the guidance for Pull Requests and requisite testing.
    List of resources for information about the Consortium and its repositories as well as model documentation, testing, and development.
+
+## License
+See our [License](Licese.pdf) and [Distribution Policy](DistributionPolicy.pdf).
```
This "License" link was broken when I looked at your rendered version.

eclare108213 replied:
fixed

apcraig commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,17 @@
+This forum is monitored by Consortium members and also opened to the whole community.
+**Please do not use our issue tracker for general support questions.**
+
+[doc-running]: https://cice-consortium-cice.readthedocs.io/en/master/user_guide/ug_running.html
```
As indicated below by @duvivier it might be good to point to the documentation on the resource page rather than directly into the readthedocs.  As with everything, I recommend we point to the resource page as much as we can, then have the resource page point out to the actual docs.

eclare108213 replied:
done

apcraig commented:
@duvivier how are you viewing these pages "rendered"?

duvivier commented:
@apcraig go directly to @phil-blain 's branch for this PR: https://github.com/phil-blain/CICE/blob/add-contributing/CONTRIBUTING.md

apcraig commented:
@duvivier makes sense, thanks.

apcraig commented:
>     * [x]  change resource index as @eclare108213 wrote here [#392 (comment)](https://github.com/CICE-Consortium/CICE/pull/392#issuecomment-578412376)
> 
>     * [x]  change FAQ @eclare108213 wrote here [#392 (comment)](https://github.com/CICE-Consortium/CICE/pull/392#issuecomment-578412376)
> 
>     * [ ]  do an icepack PR with README and CONTRIBUTING similar to the CICE ones - I can do that as soon as this one is approved **EDIT** PR is here: [CICE-Consortium/Icepack#293](https://github.com/CICE-Consortium/Icepack/pull/293) I'll update it if needed.
> 
>     * [ ]  make sure all feedback from this PR has been incorporated in the Contributing page on the wiki
> 
>     * [ ]  link to the code of conduct from the Contributing on the wiki to give it more visibility ?

I have checked off the first two boxes.

apcraig commented:
> (Note to self: update the institutional member list on the FAQ, or remove it)

As I have been advocating, I think we should document stuff in one place and then point to it as much as possible.  I know this isn't always possible nor is it always the best approach, but it makes maintenance easier.  At some point, maybe we should review our documentation and put together a little dependency/coverage diagram to help us recognize redundant documentation and/or unusual redirecting links in order to further clean this up.  I'm happy to do that if others agree, but don't want to push this point farther than I should :)

duvivier commented:
It doesn't look like this gone completed today, but we can discuss tomorrow at dinner if we want to finish it up before Tuesday.

eclare108213 commented:
I have checked the Contributing, Resource Index, About-Us and FAQ pages and further edited them as noted above.  I did not add a link to the Contributing wiki page in the main About-Us wiki page, but I put the link in the side bar in a prominent place, and generally reorganized that side bar to emphasize the important stuff.  I added a link to the code of conduct in the resources index and mentioned it in the Contributing wiki page without a direct link.

I believe that the only thing left to do is to update the Contributing.md and README.md files in this PR as suggested by @duvivier and @apcraig.  Rather than cloning the wiki and pushing to @phil-blain's branch, I am inclined to merge this PR as it is and then fix the remaining places directly.  I'll wait a few (10?) minutes before starting that process, in case @phil-blain or anyone else is working on this simultaneously.

Once that is done, we will need to bring the Icepack CONTRIBUTING.md and README.md files in line with these.
